dot/parachain/collator-protocol/message.go

```diff
@@ -177,6 +178,225 @@ const (
 	Seconded
 )
 
+// BlockedAdvertisement is vstaging advertisement that was rejected by the backing
+// subsystem. Validator may fetch it later if its fragment
+// membership gets recognized before relay parent goes out of view.
+type BlockedAdvertisement struct {
+	// peer that advertised the collation
+	peerID               peer.ID
+	collatorID           parachaintypes.CollatorID
+	candidateRelayParent common.Hash
+	candidateHash        parachaintypes.CandidateHash
+}
+
+func (cpvs CollatorProtocolValidatorSide) canSecond(
+	candidateParaID parachaintypes.ParaID,
+	candidateRelayParent common.Hash,
+	candidateHash parachaintypes.CandidateHash,
+	parentHeadDataHash common.Hash,
+) bool {
+	canSecondRequest := backing.CanSecond{
+		CandidateParaID:      candidateParaID,
+		CandidateRelayParent: candidateRelayParent,
+		CandidateHash:        candidateHash,
+		ParentHeadDataHash:   parentHeadDataHash,
+	}
+
+	responseChan := make(chan bool)
+
+	cpvs.SubSystemToOverseer <- struct {
+		responseChan     chan bool
+		canSecondRequest backing.CanSecond
+	}{
```

Comment:
Could you create a struct type for this? We will need this structure in overseer.

Reply:
This struct would be different for different messages. Type of request and response would be different most of the time.
I can create